packages/core/utils.js

```diff
@@ -13,37 +13,29 @@ export const extractComponentFromChildren = (_children, componentName) => {
   return [children, component];
 };
 
-export const buildTheme = (components, base) => {
-  // Resolve references to base theme throughout component themes
-  // eg, `link.space.mx: 2` -> `link.space.mx: '1.6rem'`
-  // eg, `link.colors.hoverColor: 'primary'` -> `link.colors.hoverColor: '#738FFC'`
+export const buildTheme = (base, components) => {
+  let baseTheme = { ...base };
 
-  let theme = { ...base };
   for (var component in components) {
     const componentTheme = components[component];
-    for (var themeCategory in componentTheme) {
-      if (["icons", "__filemeta"].includes(themeCategory)) continue;
-      // lest we want to resolve icon names with their SVG strings
+    replaceThemeRefs(componentTheme, baseTheme);
+    baseTheme[component] = componentTheme;
+  }
+  console.log("Base theme after stuff", baseTheme);
+  return baseTheme;
+};
 
-      const categoryDetails = componentTheme[themeCategory];
-      for (var key in categoryDetails) {
-        const value = categoryDetails[key];
-        const resolvedValue = base[themeCategory][value];
-        if (resolvedValue !== undefined && resolvedValue !== null) {
-          categoryDetails[key] = resolvedValue;
-        } else {
-          categoryDetails[key] = value;
-        }
+const replaceThemeRefs = (o, theme) => {
+  const keys = Object.keys(o);
+  keys
+    .filter(k => k !== "styles")
+    .forEach(k => {
+      if (theme[k]) {
+        Object.keys(o[k]).forEach(ck => {
+          o[k][ck] = theme[k][o[k][ck]] || o[k][ck];
+        });
+      } else {
+        replaceThemeRefs(o[k], theme);
```

Comment:
The recursion is smart. IIUC this will allow custom theme authors to organize their themes however they like (within the top-level layer of s-s base theme keys). This allows for intents, appearances, etc etc.